### ISSUE TITLE
Produce a set of rules to contribute to the repository

### ISSUE THREAD

**Issue by dicortazar** [chaoss/wg-diversity-inclusion#32]
After reviewing some PRs, I've realized that we should have a common way of contributing either using the command line interface or directly using the GitHub interface. Some discussion at [1].

The following is a first set of steps we should consider when preparing a PR:
* Create a fork of the repo you want to contribute to
* Create a branch. If you already have the repo, please be sure that you have upstream merged into your master branch. And then create the branch. There is some info available at [2] about syncing repositories.
* Once you feel comfortable with your new branch, please PR this. If there are a bunch of commits and those are adding things, fixing a couple of typos, etc, please consider merging some of those into a smaller set of commits. This helps to review the whole process. More info at [3].
  * If this last step sounds a bit difficult, the maintainers of the repository can help with this process when merging the PR.

Then, regarding to how to document a commit, the following are a list of ideas that help to produce an homogeneous way of working:
* Start the sentences with capital letter, write a short sentence (usually no more than ~70 characters) and use that as you use the subject of an email.
* If you want to add more information, leave a blank line
* And then start the paragraph you want to write down (please remember the limitation of 70 characters).

Others comments are welcome :)

[1] https://github.com/erlang/otp/wiki/writing-good-commit-messages
[2] https://help.github.com/articles/syncing-a-fork/
[3] https://blog.carbonfive.com/2017/08/28/always-squash-and-rebase-your-git-commits/

**Comment by GeorgLink** [comment]
Any chance we can use the CHAOSS-wide [CONTRIBUTING.md](https://github.com/chaoss/governance/blob/master/CONTRIBUTING.md) or improve it, so that we have a consistent approach throughout CHAOSS?

https://github.com/chaoss/governance/blob/master/CONTRIBUTING.md

**Comment by akshitac8** [comment]
Based on the comments can I generalise them to CONTRIBUTING.md file or we can also go with @GeorgLink  says both are really wonderful and exciting ways.
Do let me know the final say :smiley:

**Comment by GeorgLink** [comment]
To clarify: the workgroup can establish its own set of rules and procedures. However, since pull requests and GitHub always work the same, we can improve the CHAOSS-wide CONTRIBUTING.md and link to it.

**Comment by akshitac8** [comment]
Can I do that :)  ?

**Comment by GeorgLink** [comment]
​Yes, feel free to go ahead. :-)

**Comment by akshitac8** [comment]
Should I refer to this [Contributing.md](https://github.com/chaoss/governance/blob/master/CONTRIBUTING.md) file right?

**Comment by GeorgLink** [comment]
If you want to describe a generic workflow or process, then yes, the governance/CONTRIBUTING.md file is best place to refer to. 

If you want to describe a working group specific workflow or process, then we should maintain our own documentation.

**Comment by GeorgLink** [comment]
From today's discussion:

- How do we document our process and reference related works? The background of this conversation is motivated by that we build on the shoulders of giants and want to acknowledge that. Also, we want the resources that we produce to be taken up and provide evidence for why we have criteria in it.
+ We decided to put references inside the goal-question-metric files.
+ We decided to include in pull-requests rationale for our changes, to keep a documentation trail.
* TODO: describe this in our README/CONTRIBUTING file
